packages/webhooks/lib/utils.ts

```diff
@@ -115,17 +146,23 @@ export const deliver = async ({
                 { numOfAttempts: RETRY_ATTEMPTS, retry: retry.bind(this, activityLogId, logCtx) }
             );
 
-            if (activityLogId) {
+            if (logCtx) {
                 if (response.status >= 200 && response.status < 300) {
-                    await logCtx?.info(
+                    await logCtx.info(
                         `${webhookType} webhook sent successfully to the ${type} ${url} and received with a ${response.status} response code${endingMessage ? ` ${endingMessage}` : ''}.`,
                         body as Record<string, unknown>
                     );
+                    if (webhookType === 'forward') {
+                        await logCtx.success();
+                    }
                 } else {
-                    await logCtx?.error(
+                    await logCtx.error(
                         `${webhookType} sent webhook successfully to the ${type} ${url} but received a ${response.status} response code${endingMessage ? ` ${endingMessage}` : ''}. Please send a 2xx on successful receipt.`,
                         body as Record<string, unknown>
                     );
+                    if (webhookType === 'forward') {
+                        await logCtx.failed();
+                    }
```

Comment:
I would advocate for never calling failed/success deep inside methods, it's to follow the context everywhere. This function could return a boolean or a res, so you could at least do it closer to the creation

Reply:
Updated